lib/thread.c

```diff
@@ -442,8 +443,14 @@ struct thread_master *thread_master_create(const char *name)
 	rv->name = XSTRDUP(MTYPE_THREAD_MASTER, name);
 
 	/* Initialize I/O task data structures */
-	getrlimit(RLIMIT_NOFILE, &limit);
-	rv->fd_limit = (int)limit.rlim_cur;
+
```

Comment:
perhaps systemd file could also give some help on how to limit the number of fds ?
LimitNOFILE=16384

Reply:
I'm not quite sure what you're saying there: this PR doesn't prevent anyone from offering other additional ways to manage the fd set size used?

Reply:
well, I never played deeply with containers.
I just say if you run frr in docker environment using systemd, wont you have a limitation automatically detected by systemd?

otherwise, this is just a comment to understand how we can limit the max number of fds